docs/adding-a-new-ballerina-module.md

```diff
@@ -0,0 +1,482 @@
+
+This file contains the Gradle settings of the module. It includes the Gradle submodules of the module. The above mentioned directories related to `Gradle Submodule` should be added to this file using the following convention.
+
+```
+ballerina -> <module_name>-ballerina
+ballerina-tests -> <module_name>-ballerina-tests
+compiler-plugin -> <module_name>-compiler-plugin
+compiler-plugin-tests -> <module_name>-compiler-plugin-tests
+examples -> <module_name>-examples
+native -> <module_name>-native
+```
+
+This file also includes the Gradle plugins that are used to build the module.
+
+### The `spotbugs-exclude.xml` file [Optional]
+
+This file is used to skip specific spotbugs warnings/errors. This is required only if the module has Java (native) code and needs to skip specific spotbugs warnings/errors.
+
+## Adding a New Module
+
+After creating the module repository with the above structure, the following steps should be followed.
+
+### Adding the Module to the Ballerina Standard Library [Optional]
+
+This step is only required if the module is a part of the Ballerina Standard Library and it should be added to the Ballerina Standard Library dashboard and release process.
+
+To add a module to the Ballerina Standard Library, add an entry in the [`module_list.json`](https://github.com/ballerina-platform/ballerina-standard-library/blob/main/release/resources/module_list.json) file in the [Ballerina Standard Library](https://github.com/ballerina-platform/ballerina-standrad-library) Repository.
+
+>**Note:** Do not edit the `stdlid_modules.json` file manually. It will be auto-generated once the `module_list.json` file is updated.
+
+The JSON entry has two fields.
+
+#### The `name` Field
+
+This field defines the name of the module. This should be the name of the repository. (Eg.: module-ballerina-io)
+
+#### The `version_key` Field
+
+The version key of the module. This is related to the version prefix mentioned in the [`gradle.properties`](#the-gradleproperties-file-required) file. (Eg.: ballerinaStdlibIo). It is used to add the module as a dependency to another repository including the [ballerina-distribution](https://github.com/ballerina-platform/ballerina-distribution).
+
+This field is optional. It is only required if the version key cannot be inferred from the module name. (Eg.: `module-ballerina-jballerina.java.arrays` -> `stdlibJavaArraysVersion`, `module-ballerina-oauth2` -> `stdlibOAuth2Version`)
+
+>**Note:** The version key is case-sensitive and should be in the camel case format.
+
+Once the `module_list.json` file is updated, the [`Update Stdlib Dependency Graph`](https://github.com/ballerina-platform/ballerina-standard-library/actions/workflows/update_dependencies.yml) workflow will run automatically to update the Standard Library dashboard.
+
+Once updated this file, the module is ready to be released with the Ballerina Standard Library.
+
+### Adding the Module to Ballerina Daily Builds [Required]
```

Comment:
Are we talking about the Full Build Pipeline in here? If so, better to mention it properly as in some repos we seem to have a daily build workflow in the repo itself as well. (e.g. https://github.com/ballerina-platform/module-ballerinax-kafka/actions/workflows/daily-build.yml)